### PR TITLE
Update the docs handbook

### PR DESCRIPTION
**Pull request by annajung**
Signed-off-by: Anna Jung (VMware) <antheaj@vmware.com>

#### What type of PR is this:

/kind cleanup
/kind documentation

#### What this PR does / why we need it:
- Modify the localization strategy to reuse the list from the last release
- Add a section to reach out to the comms team to help with the blog posts
- Change the order around to make the release process smoother
- Remove duplicate information about creating a PR for changelogs
- Combine a few steps in the process to avoid making multiple PRs
- Add a section about validating the site during the release
- Add a section about publishing the blog post during the release

#### Which issue(s) this PR fixes:

None

#### Special notes for your reviewer:
page available for view here: https://github.com/annajung/sig-release/blob/master/release-team/role-handbooks/docs/Release-Timeline.md

/cc @jimangel @sftim @kbhawkey @savitharaghunathan @reylejano

### CODE REVIEW

File: release-team/role-handbooks/docs/Release-Timeline.md [hunk 1]
```diff
@@ -39,23 +39,23 @@ For each release, the schedule with deliverables is added to the release directo
...
     - [Review milestones](#review-milestones)
 - [Release Day](#release-day)
-    - [Merge `master`](#merge-master)
+    - [Merge the integration branch](#merge-the-integration-branch)
+    - [Publish the release blog post](#publish-the-release-blog-post)
     - [Create release with tag](#create-release-with-tag)
     - [Unfreeze](#unfreeze)
     - [Close the [future release] milestone](#close-the-future-release-milestone)
 - [Post Release Verification, Cleanup, and Handoff](#post-release-verification-cleanup-and-handoff)
-    - [Update Release Notes Changelog](#update-release-notes-changelog)
```
Comment (tengqm) [inline release-team/role-handbooks/docs/Release-Timeline.md:49]: This step is no longer required?

Comment (annajung) [reply release-team/role-handbooks/docs/Release-Timeline.md:49]: It is required but is covered by L:709.
This was a duplicate that I believe I can remove.

Comment (tengqm) [reply release-team/role-handbooks/docs/Release-Timeline.md:49]: ah, I see. Thanks for clarification.

**Pull request merged**